Aplib.Core.Tests/Desire/GoalStructureTests.cs

```diff
@@ -269,7 +326,73 @@ public void RepeatGoalStructure_WhenGoalStructureHasSucceeded_ShouldSucceed()
     }
 
     [Fact]
-    public void RepeatingGoalStructures_WhenSequenced_ShouldNotBeFinished()
+    public void RepeatGoalStructure_WhenInnerGoalStructureHasFailed_InnerGoalStructureShouldBeReset()
+    {
+        // Arrange
+        IBeliefSet beliefSet = Mock.Of<IBeliefSet>();
+        Mock<IGoalStructure<IBeliefSet>> innerGoalStructure = new();
+        innerGoalStructure.Setup(g => g.Status).Returns(CompletionStatus.Failure);
+        RepeatGoalStructure<IBeliefSet> repeatGoalStructure = new(innerGoalStructure.Object);
+
+        // Act
+        repeatGoalStructure.UpdateStatus(beliefSet);
+
+        // Assert
+        innerGoalStructure.Verify(x => x.Reset(), Times.Once);
+    }
+
+    [Fact]
+    public void RepeatGoalStructure_WhenInnerGoalStructureHasFailed_InnerGoalStructureShouldBeUnfinished()
+    {
+        // Arrange
+        Mock<IGoal<IBeliefSet>> goal = new();
+        goal.Setup(g => g.Status).Returns(CompletionStatus.Failure);
+        IBeliefSet beliefSet = Mock.Of<IBeliefSet>();
+        PrimitiveGoalStructure<IBeliefSet> primitiveGoalStructure = new(goal.Object);
+        RepeatGoalStructure<IBeliefSet> repeatGoalStructure = new(primitiveGoalStructure);
+
+        // Act
+        repeatGoalStructure.UpdateStatus(beliefSet);
+
+        // Assert
+        primitiveGoalStructure.Status.Should().Be(CompletionStatus.Unfinished);
```

Comment:
I think this should be `repeatGoalStructure`? But then still this test also has the problem that it also passes without the act section.

Reply:
It tests the inner goal structure and can be considered an integration test.